lib/indexer.rb

```diff
@@ -0,0 +1,248 @@
+# external gems
+require 'confstruct'
+
+require 'profiler/solr_doc_builder'
+require 'gdor/indexer'
+require 'solrizer'
+
+# Base class to harvest from DOR via harvestdor gem
+class Indexer < GDor::Indexer
+
+  # Initialize with configuration files
+  # @param yml_path [String] /path/to
+  # @param options [Hash]
+  def initialize(*args)
+    options = args.extract_options!
+    yml_path = args.first
+
+    @success_count = 0
+    @error_count = 0
+    @total_time_to_solr = 0
+    @total_time_to_parse = 0
+    @retries = 0
+    @druids_failed_to_ix = []
+    @validation_messages = []
+    @config ||= Confstruct::Configuration.new options
+    @config.configure(YAML.load_file(yml_path)) if yml_path && File.exist?(yml_path)
+    config.configure options
+    yield @config if block_given?
+    @harvestdor = Harvestdor::Indexer.new @config
+  end
+
+  def logger
+    config_level = case config.log_level
+      when 'debug' then Logger::DEBUG
+      when 'info' then Logger::INFO
+      when 'warn' then Logger::WARN
+      when 'error' then Logger::ERROR
+      when 'fatal' then Logger::FATAL
+    end
+    level = config_level ? config_level : Logger::INFO
+    harvestdor.logger.level = level
+    harvestdor.logger
+  end
+
+  def solr_client
+    harvestdor.solr
+  end
+
+  def metrics
+    harvestdor.metrics
+  end
+
+  # per this Indexer's config options
+  #  harvest the druids via DorFetcher
+  #   create a Solr document for each druid
+  #   write the result to the Solr index
+  #  (all members of the collection + coll rec itself)
+  def harvest_and_index(nocommit = nil)
+    nocommit = config.nocommit if nocommit.nil?
+
+    start_time = Time.now
+    logger.info("Started harvest_and_index at #{start_time}")
+
+    harvestdor.each_resource(in_threads: 3) do |resource|
+      index_with_exception_handling resource
+    end
+
+    unless nocommit
+      logger.info("Beginning Commit.")
+      solr_client.commit!
+      logger.info("Finished Commit.")
+    else
+      logger.info("Skipping commit per nocommit flag")
+    end
+
+    @total_time = elapsed_time(start_time)
+    logger.info("Finished harvest_and_index at #{Time.now}")
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.